### PR TITLE
Refine text tags

### PR DESCRIPTION
**Pull request by seoethereal**
Fix for issue #231. This essentially refines the tag types so that the tags can actually be used inside the `doctype`'s case class constructor.

### CODE REVIEW

File: scalatags/test/src/scalatags/generic/StyleSheetTests.scala [hunk 1]
```diff
@@ -86,7 +86,7 @@ abstract class StyleSheetTests[Builder, Output <: FragT, FragT]
 
   def check(txt: String, expected: String) = {
     // augmentString = work around scala/bug#11125 on JDK 11
```
Comment (lolgab) [inline scalatags/test/src/scalatags/generic/StyleSheetTests.scala:88]: If you use `linesIterator` then you can probably remove the `Predef.augmentSring` workaround and the comment.
Anyways, I would do that in a separate PR.

**Comment by lolgab** [comment]
Actually I had to do this in other tags to make the tests pass. But I didn't do for all of them so I created a bug.
Good that you did it :) thanks!

File: scalatags/test/src/scalatags/generic/StyleSheetTests.scala (hunk 1 above)
Comment (seoethereal) [reply scalatags/test/src/scalatags/generic/StyleSheetTests.scala:88]: Probably. I only replaced `lines` with `linesIterator` since it gave warnings about `lines` being deprecated or something.

**Review by lolgab — APPROVED** [review]

**Comment by lolgab** [comment]
@keynmol Thanks for the reminder. Pushed version `0.11.1`.
If there are other changes needed we can push `0.11.2` :)
